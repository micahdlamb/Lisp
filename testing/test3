(DEFUN F () (quote T))
(f)
(DEFUN J () NIL)
(J)
(DEFUN R (X) (COND((EQ X (quote NIL)) f) (NIL F) (T j)))
(DEFUN R (X) (COND((EQ X (quote NIL)) (f)) (NIL (F)) (T j)))
(R NIL)
(DEFUN PL (X Y Z) (PLUS X (MINUS Y Z)))
(PL 5 2 1)
(PL (CAR (QUOTE (5 6))) 2 1)
(DEFUN sum (X) (COND ((GREATER X 0) (plus X (sum (MINUS X 1)))) (T X)))
(sum 5)
(DEFUN likecar (a) (car a))
(likecar (quote (1 2 3)))
(DEFUN RUN (X) X)
(DEFUN SUN (X) (RUN (MINUS X 1)))
(SUN 1)
