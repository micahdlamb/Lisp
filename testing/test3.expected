F
T
J
NIL
R
R
T
PL
6
6
SUM
15
LIKECAR
1
RUN
SUN
0
